src/dns.ml

```diff
@@ -4667,23 +4668,23 @@ module Packet = struct
           in
           Header.encode buf (t.header, query, opcode, rcode);
           let names, off = Question.encode Domain_name.Map.empty buf Header.len t.question in
-          Cstruct.BE.set_uint16 buf 4 1 ;
+          Bytes.set_uint16_be buf 4 1 ;
           let names, off = encode_t names buf off t.question t.data in
           (* TODO we used to drop all other additionals if rcode <> 0 *)
           let (_names, off), adcount = encode_data t.additional names buf off in
-          Cstruct.BE.set_uint16 buf 10 adcount ;
+          Bytes.set_uint16_be buf 10 adcount ;
           (* TODO if edns embedding would truncate, we used to drop all other additionals and only encode EDNS *)
           (* TODO if additional would truncate, drop them (do not set truncation) *)
           encode_edns Rcode.NoError edns buf off, false
         with Invalid_argument _ -> (* set truncated *)
           (* if we failed to store data into buf, set truncation bit! *)
-          Cstruct.set_uint8 buf 2 (0x02 lor (Cstruct.get_uint8 buf 2)) ;
-          Cstruct.length buf, true
+          Bytes.set_uint8 buf 2 (0x02 lor (Bytes.get_uint8 buf 2)) ;
+          Bytes.length buf, true
       in
-      Cstruct.sub buf 0 off, trunc
+      String.sub (Bytes.unsafe_to_string buf) 0 off, trunc
     in
     let rec doit s =
-      let cs = Cstruct.create s in
+      let cs = Bytes.make s '\000' in
```

Comment:
I've not reviewed carefully this one, nor the following one :(

Reply:
As with `mirage-crypto`, I've been very conservative in my approach (i.e. to adopt the same behavior as `Cstruct`) because I've reported test failures due to uninitialized buffers. In my opinion, we should keep this conservative approach (and avoid a few surprises like what happened with [x509](https://github.com/mirage/mirage-crypto/pull/226)) unless we can really establish that everything is initialized correctly.

Reply:
Yes I totally approve, that's why I tried to make sure that all bytes are written when I suggested switching to `Bytes.create`. During the review I left this one, and the following one unreviewed, sorry :(
BTW I'm unsure about `let hdr = Bytes.create 12 in' in` at line 4737, as I haven't find where all bytes are initialized :(